src/js/angular/ttyg/controllers/agent-settings-modal.controller.js

```diff
@@ -1,42 +1,275 @@
+     * @return {*}
+     */
+    $scope.getNoRetrievalConnectorHelpMessage = () => {
+        // The hint contains a html link which should be properly rendered.
+        const message = decodeHTML(
+            $translate.instant(
+                'ttyg.agent.create_agent_modal.form.retrieval_search.no_retrieval_connectors_message',
+                {retrievalConnectorPage: '#/connectors'}
+            )
+        );
+        return $sce.trustAsHtml(message);
+    };
 
+    /**
+     * Closes the modal when the user confirms the agent settings and sends the payload to the parent controller opened
+     * the modal.
+     */
+    $scope.ok = () => {
+        const payload = $scope.agentFormModel.toPayload();
+        $uibModalInstance.close(payload);
     };
 
+    /**
+     * Closes the modal when the user cancels the agent creation.
+     */
     $scope.cancel = function () {
         $uibModalInstance.dismiss({});
     };
 
+    /**
+     * Closes the modal when the user clicks the close button.
+     */
     $scope.close = function () {
         $uibModalInstance.dismiss({});
     };
 
+    /**
+     * Handles the change in the repository id field. This is needed because
+     * the FTS method configuration depends on the selected repository to be
+     * able to validate if the FTS is enabled for that selected repository.
+     */
+    $scope.onRepositoryChange = () => {
+        checkIfFTSEnabled();
+    };
+
     // =========================
     // Private functions
     // =========================
 
+    const setExtractionMethodValidityStatus = () => {
+        $scope.agentSettingsForm.extractionMethods.$setValidity('required', $scope.agentFormModel.hasExtractionMethodSelected());
+    };
+
+    const checkIfFTSEnabled = () => {
+        if (!$scope.agentFormModel.repositoryId) {
+            return;
+        }
+        // pass a fake repository info object with only an id because we don't care for the location
+        RepositoriesRestService.getRepositoryModel({id: $scope.agentFormModel.repositoryId}).then((repositoryModel) => {
+            $scope.ftsEnabled = repositoryModel.getParamValue(REPOSITORY_PARAMS.enableFtsIndex);
+        })
+        .catch((error) => {
+            toastr.error($translate.instant('ttyg.agent.messages.error_repository_config_loading'));
+        })
+        .finally(() => {
+            $scope.extractionMethodLoaderFlags[ExtractionMethod.FTS_SEARCH] = false;
+        });
+    };
+
+    const handleFTSExtractionMethodPanelToggle = (extractionMethod) => {
+        $scope.extractionMethodLoaderFlags[extractionMethod.method] = true;
+        checkIfFTSEnabled();
+    };
+
+    const handleSimilaritySearchExtractionMethodPanelToggle = (extractionMethod) => {
+        $scope.extractionMethodLoaderFlags[extractionMethod.method] = true;
+        SimilarityService.getIndexesAsMenuModel().then((indexes) => {
+            $scope.similarityIndexes = indexes;
+        })
+        .catch((error) => {
```

Comment:
The error parameter is not used. It can be removed or logged to the console.

Reply:
fixed